### PR TITLE
general enhancements

### LINKED ISSUE #29: feature request: add possibility for "custom" base image
There are many options for base images in the cookie but it would be nice to also have the option of a "custom" one for matlab and opencor based services

### LINKED ISSUE #31: fix gitlab-ci.yml generated from cookiecutter
issues:
- missing call to make devenv
- deployment issues (start using sparc-external instead)
- issues with alpine based image: ```cut --fields``` not available

### PR DESCRIPTION
**Pull request by sanderegg**
## What do these changes do?

brings changes created by @mguidon and fixes found during @KZzizzle service creations

- fixes gitlab-ci.yml generation (removes staging branch, sets correct path checks, call tag-local recipe)
- fixes Makefile (tag-local recipe)
- adds custom base image
- adds centos7 base image

closes #29 
closes #31 

## Checklist

- [ ] I think the code is well written
- [ ] Unit tests for the changes exist
- [ ] Documentation reflects the changes
- [ ] If you design a new module, add your user to .github/CODEOWNERS

### CODE REVIEW

File: cookiecutter.json [hunk 1]
```diff
@@ -13,14 +13,15 @@
     "docker_base": [
         "alpine:3.7",
         "alpine:3.8",
+        "centos:7",
+        "custom:special-image",
```
Comment (pcrespov) [inline cookiecutter.json:17]: I would be a bit more thatn "special-image" here so we do not end up having "special-image1", "special-image2" :-)

File: cookiecutter.json [hunk 2]
```diff
@@ -13,14 +13,15 @@
     "docker_base": [
         "alpine:3.7",
         "alpine:3.8",
+        "centos:7",
```
Comment (GitHK) [inline cookiecutter.json:16]: 👍

File: cookiecutter.json (hunk 1 above)
Comment (GitHK) [inline cookiecutter.json:17]: what is special about this one?

Comment (sanderegg) [reply cookiecutter.json:17]: well there should be only one custom... as basically the aim is to allow to use whatever base in the dockerfile... so if we end up having a special image that becomes very trendy we would add it to the list of available images.

Comment (sanderegg) [reply cookiecutter.json:17]: it is only to ease the creation of custom images (such as based on images not available in this list, e.g. nvidia-stuff or matlab...)

Comment (GitHK) [reply cookiecutter.json:17]: Would `custom:image-template` be a better name?

Comment (KZzizzle) [reply cookiecutter.json:17]: awesome!!

**Pull request merged**